### PR TITLE
Adding secrets to the build API

### PR DESCRIPTION
**Pull request by stjohnjohnson**
Blocked on https://github.com/screwdriver-cd/models/pull/95

Feature https://github.com/screwdriver-cd/screwdriver/issues/148
Fixes https://github.com/screwdriver-cd/screwdriver/issues/163

### CODE REVIEW

File: plugins/pipelines/listJobs.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ module.exports = () => ({
     method: 'GET',
     path: '/pipelines/{id}/jobs',
     config: {
-        description: 'Get all jobs jobs for a given pipelines',
+        description: 'Get all jobs for a given pipeline',
```
Comment (d2lam) [inline plugins/pipelines/listJobs.js]: @nkatzman ^^^

File: plugins/builds/listSecrets.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                    }
+
+                    return build.secrets;
+                })
+                .then(secrets => {
+                    if (secrets.length === 0) {
+                        return reply([]);
+                    }
+
+                    return canAccess(credentials, secrets[0]).then((showSecret) =>
```
Comment (d2lam) [inline plugins/builds/listSecrets.js]: Hmmm so this doesn't take care where this build belongs to a PR job, and secrets[0] is allowed in pr, but secrets[1] is not allowed in PR, etc. ?

Comment (stjohnjohnson) [inline plugins/builds/listSecrets.js]: Don't have to, `build.secrets` are already filtered to what's available to this job, and `job.secrets` checks for PR

Comment (d2lam) [inline plugins/builds/listSecrets.js]: Oh that's right. I forgot.

**Comment by d2lam** [comment]
Cool! Looks good 👍

**Pull request merged**